tests/framework/prototype.py

```diff
@@ -586,10 +587,15 @@ def setUp(self):
     def runtest(self):
         """Run a configuration test"""
         self.cr_event.trigger()
+        self.passed = True
 
     def tearDown(self) -> None:
-        try:
-            self.cr_event.clean_up()
-        except Exception as ex:
-            logger.error(str(ex))
-            K8S_CLUSTER_MANAGER.cleanup()
+        if self.passed:
+            try:
+                self.cr_event.clean_up()
+            except Exception as ex:
+                logger.error(str(ex))
+                K8S_CLUSTER_MANAGER.cleanup()
+        else:
+            logger.info("Preserving CR in namespace %s for debugging", self.cr_event.namespace)
+            logger.info("Need to Cleanup CR manually. Use command `kubectl delete -f <yaml_file_name>.`")
```

Comment:
I believe the cleanup process should be consistent with `K8S_CLUSTER_MANAGER.cleanup()`. Currently, two types of `K8S_CLUSTER_MANAGER` are defined here: https://github.com/ray-project/kuberay/blob/master/tests/framework/utils.py, and they have different methods for cleanup.

For `KindClusterManager`, the cleanup process simply deletes the kind cluster. but for `ExternalClusterManager`, I feel there is too much manual work involved.